velox/substrait/SubstraitExtensionCollector.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/substrait/SubstraitExtensionCollector.h"
+
+namespace facebook::velox::substrait {
+
+int SubstraitExtensionCollector::getReferenceNumber(
+    const std::string& functionName,
+    const std::vector<TypePtr>& arguments) {
+  const auto& substraitFunctionSignature =
+      VeloxSubstraitSignature::toSubstraitSignature(functionName, arguments);
+  /// TODO: Currently we treat all velox registry based function signatures as
+  /// custom substrait extension, so no uri link and leave it as empty.
+  return getReferenceNumber({"", substraitFunctionSignature});
+}
+
+int SubstraitExtensionCollector::getReferenceNumber(
+    const std::string& functionName,
+    const std::vector<TypePtr>& arguments,
+    const core::AggregationNode::Step aggregationStep) {
+  // TODO: Ignore aggregationStep for now, will refactor when introduce velox
```

Comment:
Why is it OK to ignore the step? There is already a registry of aggregate functions. What's missing?

Reply:
Because we make an agreement that separate the velox registry to a new PR.